lib/govuk_publishing_components/presenters/button_helper.rb

```diff
@@ -12,7 +12,8 @@ def initialize(local_assigns)
         @title = local_assigns[:title]
         @info_text = local_assigns[:info_text]
         @rel = local_assigns[:rel]
-        @data_attributes = local_assigns[:data_attributes]
+        @default_data_attributes = { "prevent-double-click" => true }
```

Comment:
I think this should only be true if the type is unset or `submit` otherwise you could break any instances of this component where you'd want users to be able to click a button quickly.

Reply:
Good point. I'm interested to know more about the scenarios where we expect users will click two buttons quickly in sequence - that sounds more like an issue with the feature, where we would want to make a deliberate decision to allow for fast clicking in sequence.

Reply:
Fair, because of the debounce timeout..

Reply:
@benthorner: _Add item > Submit_ or _Save within a modal > Submit_

Reply:
When buttons have a type of 'button' then you can expect them to be clicked quickly.

Submit buttons already are debounced by browsers such as Internet Explorer and Edge, they're safe to debounce.

Perhaps we should add something to the guidance about this?

Reply:
@nickcolley now it's only enabled for buttons with type submit (explicitly defined or not) and added tried to document it in the yml.

Reply:
@nickcolley for the record, are you ok with how it's being applied now (submit only)?

Reply:
Can't say on the code specifics, but in theory yes.